### ISSUE TITLE
Define a continuous learning path from Fashion-MNIST to LLM training

### ISSUE THREAD

**Issue by TychoHenzen** [TychoHenzen/Pensieve#1]
## Outcome

Pensive has one authoritative, evidence-gated learning trajectory from the reported Fashion-MNIST training baseline to later LLM experiments. The trajectory inserts a bounded tiny causal-transformer decision point and prevents another large Qwen or LLM run from being treated as progress when post-training behavior is worse than pre-training behavior.

## Scope

- Reconcile `docs/PLAN.md`, `docs/STAGE-MINUS-1.md`, `docs/STAGE-MINUS-1-CONTRACT.md`, `openspec/config.yaml`, and the current Stage 0 and trainability proposals. Keep historical OpenSpec evidence intact.
- Distinguish code-complete, trainability-demonstrated, and gate-passed states. A stage must not be marked complete from code or unit tests alone.
- Define a continuous ladder that starts with the protected Fashion-MNIST trainability baseline, evaluates a tiny causal transformer trained from scratch, moves to a small held-out autoregressive task, applies one controlled latent modification, and reaches a larger LLM task only after the earlier gates pass. The plan may select a different ladder, but it must record the evidence and rationale.
- For every stage, name the model and size, data or task, objective, matched baseline, pre-training and post-training metrics, seed and compute budget, pass condition, and stop or retreat condition.
- Define the tiny-transformer milestone well enough for a later implementation ticket. It must specify the exact task, model scale, tokenization, training split, checkpoint policy, and numeric gates for fixed-set overfit, held-out behavior, and regression detection.
- Define a regression rule that blocks advancement when task behavior, output validity, diversity or representation separation, retention, or reproducibility worsens, even if a training loss improves.

Non-goals: implement the transformer, train Qwen or another LLM, resume the failed full run, start the Stage 1 recurrent backbone, add fast weights or consolidation, weaken existing Stage -1 or Stage 0 gates, or delete historical documents and artifacts.

## Implementation notes

- `core/latent_loop.py:20-64` hard-wires the production latent loop to a frozen `Qwen/Qwen2.5-0.5B-Instruct` backbone, requires hidden width 896, and trains only wrapper projections around it. `LatentLoop.run()` performs a fixed number of latent steps and uses Qwen-specific hidden-state tapping.
- `train/trainer.py:1-12,51-141` freezes Qwen and MiniLM and updates encoder and latent-loop wrapper parameters with a teacher-forced answer loss, prompt-alignment penalty, and collapse penalty. This is not general end-to-end language-model training, so loss movement cannot stand in for useful decoded behavior.
- `train/run_alternating.py:503-588,755-915` runs alternating Gradient and EGGROLL updates on the fixed Calc-ASDiv_A path with 570 training records, held-out evaluation, a Qwen-specific stability report, and Qwen plus MiniLM construction. It is not a small transformer learning baseline.
- `eval/gate/token_cot_baseline.py` and `eval/gate/latent_eval.py` define a Qwen-specific 520-item test selection, pinned prompt and asset identity, versioned checkpoint loading, and five-seed latent evaluation. That gate cannot directly answer whether a tiny transformer has learned basic causal modeling.
- `train/run_stage0_trainability.py:82-89` currently returns `inconclusive` with `investigation_incomplete` instead of running the bounded probes. The untracked `scripts/training_validation/03-diagnose-failed-stability.ps1:47-52` correctly refuses to continue when the report contains no probe evidence. The new trajectory must treat this investigation as a prerequisite, not as completed proof.
- `docs/PLAN.md:47-89` jumps from the Stage -1 harness to the latent Stage 0 and then to a Mamba or xLSTM Stage 1. `docs/STAGE-MINUS-1.md:1-18,300-306` still describes the full five-seed Split-MNIST reproduction as remaining, while `openspec/config.yaml:9` says Stage -1 is complete and `CLAUDE.md` says Stage 0 is complete in code only. The refinement must reconcile these claims without confusing implementation status with measured success.
- A repository search found no tiny-transformer implementation and no Fashion-MNIST trainer in the current checkout. The current MNIST code is Stage -1 split-classify data binding. The roadmap must identify the reported baseline's reproducible command, configuration, and result, or explicitly make baseline recovery the first step.
- `CLEANUP_LEDGER.md` records unresolved Stage -1 instrumentation and run-record gaps and the incomplete trainability runner. Those are evidence boundaries for the roadmap, not reasons to start a larger model run.
- The implementation direction is to isolate basic transformer training from the Qwen hidden-state tap, MiniLM encoder, latent slots, decoder, and numerical gate. This makes a tiny-transformer failure attributable to the transformer training path rather than to several coupled Stage 0 components.

Priority: `Prio 2 - Urgent`. Its live description covers serious issues or important milestones that heavily impact progress. The current path couples a large frozen backbone, latent wrappers, and an unsuitable problem task, while the available trainability command still produces no probe evidence. Replanning is required before another expensive run can provide useful evidence.

Effort: `Effort 5 - Large`. Its live description covers a complex task that requires significant effort or has notable dependencies. This work spans several roadmap and contract documents, baseline provenance, training entry points, and evaluation boundaries, but excludes model implementation and remains below the epic threshold.

Classification: `documentation`. The deliverable is an authoritative roadmap and status reconciliation. The tiny-transformer implementation is a later feature and is explicitly outside this PBI.

## Acceptance criteria

- [ ] `docs/PLAN.md` is the authoritative roadmap and contains an ordered Fashion-MNIST-to-LLM ladder. Every stage records model scale, task or data, objective, matched baseline, pre/post metrics, budget, gate, and stop or retreat rule.
- [ ] The affected Stage -1 and Stage 0 documents distinguish code, trainability evidence, and gate completion. Cross-links and `openspec/config.yaml` no longer present contradictory completion claims.
- [ ] The reported Fashion-MNIST routine is named as a protected baseline with reproducible provenance. If it is absent from the checkout, the first recovery step and its evidence requirement are explicit.
- [ ] The roadmap explicitly selects or rejects a tiny causal transformer as the next bridge. If selected, it defines a deterministic task, model scale, train and held-out data, checkpoint and seed policy, and numeric gates for tiny-set overfit, held-out behavior, valid output, and pre/post regression.
- [ ] Advancement is blocked when behavior regresses, output becomes invalid or collapsed, retention worsens, or repeatability fails, regardless of loss reduction.
- [ ] The change does not implement or launch a model run. Existing OpenSpec validation and current Stage 0 behavior remain intact.

## Verification

- `git diff --check`
- `openspec validate --all --strict --no-interactive`
- `\.venv\Scripts\python.exe -m pytest -q -p no:cacheprovider tests/test_latent_loop.py tests/test_latent_core_subject.py tests/test_gradient_training.py tests/test_run_stage0_trainability.py tests/gate/test_stage0_latent_evaluation.py tests/gate/test_stage0_token_baseline.py`
- Read the affected roadmap and contract documents and verify each stage has the required fields, one source of truth exists for status, and every cross-link resolves.
- Verify the diff contains only the intended documentation and planning changes, with no new training outputs, model dependencies, or model implementation.
